### PR TITLE
Add pairplot method to visualize correlation between time series

### PR DESCRIPTION
**Pull request by martinvonk**
# Short Description
Add pairplot method to investigate correlation between series. 

Inspired by the seaborn pairplot method but this function is simpler and adds the correlation coefficient.

image

# Checklist before PR can be merged:
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed
- [x] Example Notebook (for new features)
- [x] Remove output for all notebooks with changes

### CODE REVIEW

**Comment by martinvonk** [comment]
I haven't made an example (notebook) yet. Maybe someone has a good idea where to show this?

**Comment by dbrakenhoff** [comment]
> I haven't made an example (notebook) yet. Maybe someone has a good idea where to show this?

Perhaps, I can include this plot when i complete the notebook about cross-correlation?

File: pastas/modelcompare.py [hunk 1]
```diff
@@ -854,31 +854,6 @@ def plot_table_diagnostics(
         cols = diags.columns.to_list()[-1:] + diags.columns.to_list()[:-1]
         return self.plot_table(axn=axn, df=diags[cols])
 
-    def share_xaxes(self, axes: List[Axes]) -> None:
```
Comment (dbrakenhoff) [inline pastas/modelcompare.py:857]: I'd vote for leaving this method but using the new function share_xaxes().

File: tests/test_comparison.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def test_comparison_plot_custom(ml: ps.Model, ml_sm: ps.Model) -> None:
     mc.plot_contribution(axn="con{i}")
     mc.plot_table_diagnostics(axn="dia", diag_col="Statistic")
     mc.plot_acf(axn="acf")
-    mc.share_xaxes(
+    ps.plots.share_xaxes(
```
Comment (dbrakenhoff) [inline tests/test_comparison.py:36]: See earlier comment, I would keep `mc.share_xaxes()`

**Review by dbrakenhoff — APPROVED** [review]

File: pastas/modelcompare.py (hunk 1 above)
Comment (martinvonk) [reply pastas/modelcompare.py:857]: Agreed.

**Comment by martinvonk** [comment]
Nice to have `model.plots.pairplot()`, similar to `model.plots.series()` which relies on the `pastas.plots.series()` function.

**Comment by martinvonk** [comment]
Possibly move to preprocessing library if that exists.

**Comment by raoulcollenteur** [comment]
This PR is missing an Issue that it solves, perhaps open one. 

I forgot what we decided on this feature? Add it or not?

**Comment by raoulcollenteur** [comment]
Related to #350 ?

**Comment by martinvonk** [comment]
Partially related to #350 but this PR has no specific issue. We would add this PR and maybe move it if there exists a Pastas preprocessing library.

**Pull request merged**